rotkehlchen/api/v1/schemas.py

```diff
@@ -806,9 +806,12 @@ def make_history_base_entry(
                 data: dict[str, Any],
                 **_kwargs: Any,
         ) -> dict[str, Any]:
-            if data['location'] == Location.BITCOIN and data['asset'] != A_BTC:
+            if (
+                ((location := data['location']) == Location.BITCOIN and data['asset'] != A_BTC) or
+                (location == Location.BITCOIN_CASH and data['asset'] != A_BCH)
+            ):
                 raise ValidationError(
-                    message='Bitcoin events must use BTC as the asset',
+                    message=f'{location.name.lower()} events must use the native asset',
```

Comment:
Isn't there a way to get the native asset (BTC or BCH) from the lcoation and then also show it here in the user facing error?